torch/autograd/_functions/linalg.py

```diff
@@ -102,3 +102,42 @@ def backward(ctx, grad_output, grad_LU=None):
         grad_b, _ = torch.gesv(grad_output, a.t())
         grad_a = -torch.mm(grad_b, X.t())
         return grad_b, grad_a
+
+class Symeig(Function):
+
+    @staticmethod
+    def forward(ctx, input, eigenvectors=False, upper=True):
+        ctx.eigenvectors = eigenvectors
+        ctx.upper = upper
+        w, v = torch.symeig(input, eigenvectors=ctx.eigenvectors, upper=ctx.upper)
+        ctx.save_for_backward(input, w, v)
+        return w, v
+
+    @staticmethod
+    def backward(ctx, grad_w, grad_v):
+        x, w, v, = ctx.saved_variables
+
+        # gives an error if I don't do this..
+        x = x.data
```

Comment:
This doesn't look right. If you need tensors in your backward, you should use the devorator `once_differentiable` in the function, but looking at the functions used, that shouldn't be necessary.